### DISCUSSION TITLE
How to configure iperf-3.1.3-win64 to generate 99.5% throughput traffic on 1Gbps Ethernet?

### DISCUSSION THREAD

**Discussion by lksark** [esnet/iperf#1625] [Q&A] [answered]
I would like to ask how can we configure iperf-3.1.3-win64 to generate 99.5% or 995 Mbits/sec throughput traffic? We managed to get ~943 Mbits/sec traffic on 1Gbps Ethernet.

Our computers are running on Microsoft Windows 10 Pro with 16GB RAM.

![iperf-3 1 3-win64 traffic](https://github.com/esnet/iperf/assets/26447759/ec684247-0fb3-44a9-a93e-dc813801dabf)

**Comment by davidBar-On** [comment, marked as answer]
Hi, 0.94Gbps on a 1Gbps interface seems quite good.  Also, the bottleneck may be in the network cards, network buffers, etc, and not in iperf3.

Regarding iperf3, using UDP (`-u`) may allow increasing the throughput, but for that to work you need to limit the sent throughput to 1Gbps (`-b `G`) or a little bit more and try different packet sizes.  Note that TCP default packet is 128KB and UDP default is 1460 bytes, so try different sizes between these values (using the `-l` option).  For UDP, the throughput is measured on the receiver since packets may be lost.

You can also try setting different TCP packets sizes - greater or smaller than the default 128KB.

Another thing that may help is running few streams in parallel (`-P` option) - TCP or UDP.   In this case, the cumulative streams throughput may be higher then a single stream.  Note that when using multi-streams the throughput set by `-b` is for each stream, so e.g. for UDP test using 4 streams (`P 4`) set throughput to 250KB (`-b 250KB`). 

One more option that can help is the burst count.  You may try setting it to 20 or 30 (default is 10).  This is done using the `-b` option: to set burst count to 20 use for TCP use `-b /20` and for UDP `-b 1G/20`.

Note that iperf3.1.3 is very old, and probably not all the options are supported.  It is better to upgrade to a new version (can be downloaded from [here](https://files.budman.pw/)  - not an official site).  Also, iperf3.16 is multi-threaded which may help with the parallel streams (`-P`) tests.
